utils/src/main/java/org/jclouds/karaf/utils/EnvHelper.java

```diff
@@ -238,4 +240,12 @@ private static String getValueOrPropertyOrEnvironmentVariable(
         }
         return value;
     }
+    
+   private static String getCredentialValue(String provider, String credential) {
+      return credential != null && isGoogleCloud(provider) ? getGoogleCredentialFromJsonFileIfPath(credential) : credential;
+   }
+   
+   private static boolean isGoogleCloud(String provider) {
+      return provider != null && provider.startsWith("google");
```

Comment:
Yes, it can be null if the user hasn't provided explicitly the parameter or the environment variable. For example, when reusing stored compute service instances, the user may just provide the registered name instead of all the information.

The code that creates the services first reads all parameters and performs some validations or others based on the ones that have been set , then instantiates the services accordingly.